Foundation/URLSession/URLSessionTask.swift

```diff
@@ -252,8 +336,10 @@ open class URLSessionTask : NSObject, NSCopying {
             self.updateTaskState()
             
             if self.suspendCount == 1 {
-                self.workQueue.async {
-                    self._protocol?.stopLoading()
+                self._getProtocol { (urlProtocol) in
+                    self.workQueue.async {
```

Comment:
It is complicated to determine, but with the `workQueue` and the protocol being invalidated inside `_protocolLock`, some of these pieces of code might not have the same behaviour and might not be expected. If `_getProtocol` here returns a non-`nil` protocol, it might have already been invalidated by the time the `workQueue` executes the code. That didn't happen in the previous version.

I don't know if that's important or not, however.